src/service/video-manager-impl.js

```diff
@@ -284,20 +303,51 @@ export class VideoManager {
     return this.getEntryForVideo_(video).userInteractedWithAutoPlay();
   }
 
+  /**
+   * Checks whether there's no video already docked
+   *
+   * @param {VideoEntry} entry
+   * @return {boolean}
+   */
+  canDock(entry) {
+    return !this.dockedVideo_ || this.dockedVideo_ == entry;
+  }
+
+  /**
+   * Registers the provided video as docked
+   *
+   * @param {VideoEntry} entry
+   */
+  registerDocked(entry) {
+    this.dockedVideo_ = entry;
+  }
+
+  /**
+   * Un-registers the currently docked video
+   */
+  unregisterDocked() {
+    this.dockedVideo_ = null;
+    for (let i = 0; i < this.entries_.length; i++) {
+      this.entries_[i].hasBeenInViewBefore = false;
```

Comment:
this doesn't make sense to me. also this variable is only "dock-related". Can you think of a more descriptive name for this?